independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/VertxRedirectHandler.java

```diff
@@ -0,0 +1,13 @@
+package org.jboss.resteasy.reactive.client.handlers;
+
+import java.util.function.Function;
+
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.RequestOptions;
+
+/**
+ * This handler is invoked when target server returns an HTTP status of family redirection.
+ */
+public interface VertxRedirectHandler extends Function<HttpClientResponse, Future<RequestOptions>> {
```

Comment:
The interface RedirectHandler is the preferred one as users do not need to know what the internal HTTP client actually is. For this reason, this option is the only one I documented. However, I thought it could also be good to give users the option to use the Vert.x-based solution.

And thinking again about this, if we add this class, we had to maintain it in case of Vert.x is dropped, so maybe it's better to remove it. wdyt? @geoand

Reply:
I'd like to avoid advertisting a Vert.x specific solution. It's fine for us or other layered products, but not really good for users

Reply:
+1 removing it

Reply:
PR updated.